src/main/java/com/amazon/opendistroforelasticsearch/security/securityconf/ConfigModelV7.java

```diff
@@ -743,8 +745,10 @@ public String getUnresolvedIndexPattern(User user) {
             WildcardMatcher matcher = WildcardMatcher.from(unresolved);
             String[] resolved = null;
             if (!(matcher instanceof WildcardMatcher.Exact)) {
-                final String[] aliasesForPermittedPattern = cs.state().getMetaData().getAliasAndIndexLookup().entrySet().stream()
-                        .filter(e -> e.getValue().isAlias()).filter(e -> matcher.test(e.getKey())).map(e -> e.getKey())
+                final String[] aliasesForPermittedPattern = cs.state().getMetadata().getIndicesLookup().entrySet().stream()
+                        .filter(e -> e.getValue().getType() == ALIAS)
```

Comment:
nit: is ALIAS a string? `.equals` comparison may be better here.

Reply:
`ALIAS` is enum.